vlayer/ethereum/circuit/lib/src/nft.nr

```diff
@@ -0,0 +1,22 @@
+use dep::ethereum_history_api::{
+    account_with_storage::get_account_with_storage,
+    misc::types::{Address, Bytes32, ADDRESS_LENGTH, BYTES32_LENGTH}, misc::arrays::subarray
+};
+
+struct ERC721Token {
+    address: Address,
+    token_id_to_slot: fn (Bytes32) -> Bytes32,
+    chain_id: Field
+}
+
+trait ERC721 {
+    fn get_owner(self, token_id: Bytes32, block_number: Field) -> Address;
+}
+
+impl ERC721 for ERC721Token {
+    fn get_owner(self, token_id: Bytes32, block_number: Field) -> Address {
+        let storage_key = (self.token_id_to_slot)(token_id);
+        let account = get_account_with_storage(block_number, self.address, storage_key);
```

Comment:
`(values, _, _, )`

Reply:
Ok, changed. But IMO previous one is more concise. Just for comparison:
```rust
let StorageWithinBlock { block_hash: _, account: _, values } = get_account_with_storage(block_number, self.address, storage_key);
subarray(values[0], BYTES32_LENGTH - ADDRESS_LENGTH)
```
vs 
```rust
let storage_within_block = get_account_with_storage(block_number, self.address, storage_key);
subarray(storage_within_block.values[0], BYTES32_LENGTH - ADDRESS_LENGTH)
```